### PR TITLE
Feature/cfg 287 improve handling of invalid date in mz datetime

### PR DESCRIPTION
**Pull request by Paul-Free**

### CODE REVIEW

File: projects/components/src/lib/date-time/date-time.component.ts [hunk 1]
```diff
@@ -5,96 +5,150 @@ import { TimeValue } from './time-value';
...
-    writeValue(value: Date) {
-        if (value) {
-            this.date = value;
-            const hourPrefix = value.getHours() > 9 ? '' : '0';
-            const minutePrefix = value.getMinutes() > 9 ? '' : '0';
-            this.time = `${hourPrefix}${value.getHours()}:${minutePrefix}${value.getMinutes()}`;
-            this._cd.markForCheck();
-        }
-    }
+  date: Date | undefined | null;
```
Comment (ruifang) [inline projects/components/src/lib/date-time/date-time.component.ts]: The tab width in this PR looks different. Do you have prettier extension installed? We'll want to use that to format files in this project.

**Review by ruifang — CHANGES_REQUESTED** [review]

File: projects/components/src/lib/date-time/date-time.component.ts [hunk 2]
```diff
@@ -5,96 +5,150 @@ import { TimeValue } from './time-value';
...
+  time: string | undefined | null;
+  private tempDate: string | undefined | null = null;
+  private tempTime: string | undefined | null = null;
```
Comment (ruifang) [inline projects/components/src/lib/date-time/date-time.component.ts]: What are the tempDate and tempTime for?

**Review by ruifang — CHANGES_REQUESTED** [review]

File: projects/components/src/lib/date-time/date-time.component.ts (hunk 2 above)
Comment (ruifang) [reply projects/components/src/lib/date-time/date-time.component.ts]: Looks like they were already here, so ignore my question.

**Review by eb-g3dg5 — APPROVED** [review]

File: projects/components/src/lib/date-time/date-time.component.spec.ts [hunk 3]
```diff
@@ -5,89 +5,78 @@ import { MatNativeDateModule } from '@angular/material/core';
...
+            tick();
+            fixture.detectChanges();
+            expectToDisplayDateTime(fixture.nativeElement, '13:50', '11/12/2022');
+
+            fixture.componentInstance.dateTime = new Date('09/11/2022 09:10');
+            fixture.detectChanges();
+            tick();
+            fixture.detectChanges();
+            expectToDisplayDateTime(fixture.nativeElement, '09:10', '8/11/2022');
+        }));
```
Comment (SubjectiveReality) [inline projects/components/src/lib/date-time/date-time.component.spec.ts:172]: a new test case or two could be added, like when deleting part of the datetime

File: projects/components/src/lib/date-time/date-time.component.spec.ts [hunk 4]
```diff
@@ -5,89 +5,170 @@ import { MatNativeDateModule } from '@angular/material/core';
 import { DateTimeComponent } from './date-time.component';
 import { MuziehDateTimeModule } from './date-time.module';
 
-
-@Component({selector: 'test-cmp', template: ''})
+@Component({ selector: 'test-cmp', template: '' })
 class TestComponent {
-  dateTime: Date | undefined;
+    dateTime: Date | undefined | null;
```
Comment (SubjectiveReality) [inline projects/components/src/lib/date-time/date-time.component.spec.ts]: no null, just undefined as discovered in TS docs https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined

File: projects/components/src/lib/date-time/date-time.component.ts [hunk 5]
```diff
@@ -17,10 +17,25 @@ import { MatDatepicker } from '@angular/material/datepicker';
     ],
 })
 export class DateTimeComponent implements ControlValueAccessor {
-    date: Date | undefined;
-    time: String | undefined;
-    private tempDate: string | undefined | null = null;
-    private tempTime: string | null = null;
+    date: Date | undefined | null;
+    time: string | undefined | null;
```
Comment (SubjectiveReality) [inline projects/components/src/lib/date-time/date-time.component.ts]: revert these to just undefined no null, per TS docs https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined

File: projects/components/src/lib/date-time/date-time.component.spec.ts (hunk 4 above)
Comment (SubjectiveReality) [reply projects/components/src/lib/date-time/date-time.component.spec.ts]: done

File: projects/components/src/lib/date-time/date-time.component.ts (hunk 5 above)
Comment (SubjectiveReality) [reply projects/components/src/lib/date-time/date-time.component.ts]: done

**Review by SubjectiveReality — APPROVED** [review]

**Review by ruifang — APPROVED** [review]

**Pull request merged**